### ISSUE TITLE
One broken stub in a fragment file causes the whole file to not load

### ISSUE THREAD

**Issue by PankajBhojwani** [microsoft/terminal#11330]
### Windows Terminal version (or Windows build number)

Current main build

### Steps to reproduce

Put the following fragment file in `AppData\Local\Microsoft\Windows Terminal\Fragments\{any-name-here}`:

```
{
  "profiles": [
    {
      "name": "broken profile",
      "font": "thisFontDoesntExist"
    },
    {
      "name": "Cool Profile"
    }
  ]
}
```

Then start WT

### Expected Behavior

"Cool Profile" should show up in the dropdown

### Actual Behavior

"Cool Profile" does not show up, the presence of the broken profile causes the non-broken profile to not load

**Comment by lhecker** [comment]
This behavior is consistent with the previous behavior. You can try it with:
```json
{
    "profiles": [
        {
            "name": "broken profile",
            "unfocusedAppearance": ""
        },
        {
            "name": "Cool Profile"
        }
    ]
}
```

I'll tag this issue as needs-discussion, as we'd have to decide what the new failure handling would need to look like. After all, errors in the user's settings.json doesn't just silently ignore such invalid profiles either. A warning might be appropriate here...

**Comment by PankajBhojwani** [comment]
I have a branch pre-#11184 and with that example, `"Cool Profile"` loads while `"broken profile"` does not, so I think this was a recent regression! In #10601 we added try-catch blocks around the _individual_ profile and update stubs within a fragment file to ensure we continue parsing fragments even if we find broken ones, which I believe the new implementation is missing.

We decided to silently ignore issues in fragment parsing because fragment issues are rarely the fault of the user, so we didn't think there was any point in alerting the user about that. Totally down to brainstorm other ways to avoid silently failing though
